torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2019,12 +2021,12 @@ def _wait_for_previous_optim_step(self) -> None:
         """
         if not self._is_root:
             return
-        current_stream = torch.cuda.current_stream()
-        self._streams["unshard"].wait_stream(current_stream)
+        computation_stream = self._streams["computation"]
```

Comment:
Wouldn't this break running FSDP under non-default stream? I am not sure if that works at the moment, though.

Basically, with this PR we save current stream in "computation" in _init_stream, and then use that stream in this _wait_for_prev_optim_step function. But if somehow user changed current stream in between these 2 functions, we'd change behavior in that case?